src/MetalsFeatures.ts

```diff
@@ -4,20 +4,17 @@ import {
 } from "vscode-languageserver-protocol";
 import { StaticFeature, workspace } from "coc.nvim";
 
-export interface TreeViewProvider {}
```

Comment:
Because I defined `TreeViewFeature` in separate file tvp/feature.ts. File tvp/features.ts is quite big so I don't think we should define all features in one file. Debugging feature will probably require even more notifications than TreeViewProtocol.

File language-client/client.ts from coc.nvim also define a lot of features and they are defined in different classes. So I think we should follow the same pattern. What do you think?

Reply:
Hmm, yea, I guess I'm open to it. Looking through it, I guess they don't absolutely need to be together, and if the VS Code extension it's easy to since there really isn't much else that goes into it, but seeing that this one is quite involved, maybe it makes sense. I also think seeing how this one is done, I could also move the decoration protocol one to be a bit more similar to this as well.